src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/HostDiagnosticAnalyzerProvider.cs

```diff
@@ -8,44 +8,17 @@
 using Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
-internal sealed class HostDiagnosticAnalyzerProvider : IHostDiagnosticAnalyzerProvider
-{
-
-    private readonly ImmutableArray<(AnalyzerFileReference reference, string extensionId)> _analyzerReferences;
 
-    public HostDiagnosticAnalyzerProvider(string? razorSourceGenerator)
-    {
-        if (razorSourceGenerator == null || !File.Exists(razorSourceGenerator))
-        {
-            _analyzerReferences = [];
-        }
-        else
-        {
-            _analyzerReferences = [(
-                new AnalyzerFileReference(razorSourceGenerator, new SimpleAnalyzerAssemblyLoader()),
-                ProjectSystemProject.RazorVsixExtensionId
-            )];
-        }
-    }
-
-    public ImmutableArray<(AnalyzerFileReference reference, string extensionId)> GetAnalyzerReferencesInExtensions()
-    {
-        return _analyzerReferences;
-    }
+internal sealed class HostDiagnosticAnalyzerProvider(string? razorSourceGenerator) : IHostDiagnosticAnalyzerProvider
+{
+    public ImmutableArray<(AnalyzerFileReference reference, string extensionId)> GetAnalyzerReferencesInExtensions() => [];
 
-    private sealed class SimpleAnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
+    public ImmutableArray<(string path, string extensionId)> GetRazorAssembliesInExtensions()
     {
-        public void AddDependencyLocation(string fullPath)
-        {
-            // This method is used to add a path that should be probed for analyzer dependencies.
-            // In this simple implementation, we do nothing.
-        }
-
-        public Assembly LoadFromPath(string fullPath)
+        if (razorSourceGenerator is not null)
```

Comment:
The previous code had a `File.Exists` call, is it okay to remove that here? Also are we sure this points to the generator DLL and not the generator directory?

Reply:
Basically the old code, intentionally or accidentally, guarded against a directory being specified with the `File.Exists` call. That is removed now. So if someone is passing the razor generator directory in a VS code setting what will happen now?

Reply:
Ah yeah, good catch, will put the exists back in.